### PR TITLE
Add case object in `Value`

### PR DESCRIPTION
**Pull request by ashalva**
## Motivation
Currently Toggles support only primitive types, but seemingly there is some need to add support for objects. I will be working for some time to add support for objects in Toggles. So, expect some more prs in upcomings days/weeks.

I have introduced the `Object` in https://github.com/TogglesPlatform/Toggles/pull/36. Now its time to actually using and add support object types in Toggles 🚀 

## Implementation

For the memory this is how the object based Toggle will be represented as json 👇 

<details>
<summary>Object json</summary>

```json
{
    "variable": "object_toggle",
    "object": {
            "boolProperty" : true,
            "stringProperty" : "value",
            "intProperty" : 421,
            "numberProperty" : 12.3,
    },
    "metadata": {
        "group": "Raw toggles",
        "description": "Object toggle"
    }
}
```

</details>

From now on `Object` and `ObjectSupportedType` both conform `Codable`(instead of `Decodable`) and `Equatable`. I had to conform to that since `Value` enum is `Codable & Equatable`.

Added new case `object(Object)` in `Value` which forced me to consider new case in multiple places:

1. `Toggle+Codable.swift` - Decodes json to one of the cases of `Value`.
2. `Value+Utilities.swift` - Provides utilities for `Toggle` e.g description, icon etc. 
    1. In order to display object toggle visually nicely, I have added new property `description` in `Object` which converts `map` into pretty json string (Please see attached video)
4. `InputValidationHelper.swift` - Validates the input value for overrides. Considering now `.object` as well. Created two extensions to fulfill my needs:
    1. `String+ValidJSON.swift` - validates if input string can be decoded in `Object`.
    2. `String+Object.swift` - tries to convert input string to `Object`. 

This is how it looks in Demo app:

<video width='200' src='https://github.com/TogglesPlatform/Toggles/assets/9494262/717822ea-0576-4587-9743-129d1a995d89'>

## What is Next?

I am not including `ToggleObservable` changes on purpose as the pr is getting large. So expect another pr from me to cover that part as well.

### CODE REVIEW

**Review by albertodebortoli — APPROVED** [review]
Beautiful PR @ashalva! Thanks 👏

File: Tests/Suites/Extensions/Value+AccessorsTests.swift [hunk 1]
```diff
@@ -13,6 +13,7 @@ final class Value_AccessorsTests: XCTestCase {
         XCTAssertNil(factory.numericToggle(3.1416).value.boolValue)
         XCTAssertNil(factory.stringToggle("Hello World").value.boolValue)
         XCTAssertNil(factory.secureToggle("secret").value.boolValue)
+        XCTAssertNil(factory.objetToggle(Object(map: ["var": .int(333)])).value.boolValue)
```
Comment (AlexL271828) [inline Tests/Suites/Extensions/Value+AccessorsTests.swift:16]: *`objectToggle`

File: Tests/Suites/Utilities/InputValidationHelperTests.swift [hunk 2]
```diff
@@ -61,6 +61,18 @@ final class InputValidationHelperTests: XCTestCase {
         XCTAssertTrue(inputValidationHelper.isInputValid("***"))
     }
     
+    func test_inputValidObjectTrue() throws {
+        let toggle = factory.objetEmptyMetadataToggle(Object(map: ["var": .int(300)]))
```
Comment (AlexL271828) [inline Tests/Suites/Utilities/InputValidationHelperTests.swift:65]: *`objectEmptyMetadataToggle`

File: Tests/Utilities/ToggleFactory.swift [hunk 3]
```diff
@@ -64,13 +70,25 @@ class ToggleFactory {
...
+    
     func makeToggles() -> [Toggle] {
         [
             booleanToggle(true),
             integerToggle(42),
             numericToggle(3.1416),
             stringToggle("Hello World"),
-            secureToggle("YXe+Ev76FbdwCeDCVpZNZ1RItWZwKTLXF3/Yi+x62n3JWbvPo6YK")
+            secureToggle("YXe+Ev76FbdwCeDCVpZNZ1RItWZwKTLXF3/Yi+x62n3JWbvPo6YK"),
+            objetToggle(Object(map: [
```
Comment (AlexL271828) [inline Tests/Utilities/ToggleFactory.swift:86]: maybe worth adding this to the other makeToggles functions? i.e `makeTogglesWithNoMetadata`, `makeToggles(count: Int) `

**Review by AlexL271828 — APPROVED** [review]

File: Tests/Utilities/ToggleFactory.swift (hunk 3 above)
Comment (ashalva) [reply Tests/Utilities/ToggleFactory.swift:86]: Which other `makeToggles` functions do you mean? I have added two functions `objetEmptyMetadataToggle` and `objetToggle`, anything I missed?

File: Tests/Suites/Extensions/Value+AccessorsTests.swift (hunk 1 above)
Comment (ashalva) [reply Tests/Suites/Extensions/Value+AccessorsTests.swift:16]: Nice catch!

File: Tests/Utilities/ToggleFactory.swift (hunk 3 above)
Comment (ashalva) [reply Tests/Utilities/ToggleFactory.swift]: Resolving conversation as pr updated ✅

**Pull request merged**